### DISCUSSION TITLE
exe 파일 배포 후 grpc 동작 안되는 문제

### DISCUSSION THREAD

**Discussion by ysmanse** [elbakramer/koapy#93] [Q&A]
안녕하세요, 먼저 좋은 소스 공개해주셔서 너무 유용하게 잘 쓰고 있습니다. 

koapy 를 활용해서 매매에 도움이 될만한 프로그램을 만들었고, 이 프로그램을 지인들과 함께 사용하려고 합니다.
지인들은 코딩, python에 관해 전혀 무지한 사람들이라 소스를 주고 실행을 해보라고 할 수 없는 상황인지라
pyinstaller를 통해 개발한 응용프로그램을 exe 파일로 빌드 하였습니다. 빌드 후 실행을 해보니

`"[DEBUG] Creating manager application - KiwoomOpenApiPlusManagerApplication.py:207"`

메세지 이후 

```
File "koapy\backend\kiwoom_open_api_plus\pyside2\KiwoomOpenApiPlusManagerApplication.py", line 821, in main
  File "koapy\backend\kiwoom_open_api_plus\pyside2\KiwoomOpenApiPlusManagerApplication.py", line 293, in __init__
  File "koapy\backend\kiwoom_open_api_plus\pyside2\KiwoomOpenApiPlusManagerApplication.py", line 361, in _reinitializeServerProcessAndGrpcClient
AssertionError: Client is not ready
[40564] Failed to execute script 'main' due to unhandled exception!
QProcess: Destroyed while process ("C:\\Users\\choi\\Documents\\workspace\\KoapyServer\\dist\\main.exe") is still running.
```

타임아웃을 맞고 앱이 종료되어 버립니다.

소스코드로 실행할 때는 매우 잘 동작하던 녀석이 pyinstaller를 통해 exe 빌드를 하면 grpc 서버를 올리지 못하는것 같습니다.
이 외에는 에러메제지가 없어 이슈 트래킹을 못하는 상황이라 혹시 짐작 가시는 부분이 있다면 도움 부탁드립니다.

감사합니다!

**Comment by elbakramer** [comment]
안녕하세요. 

가능하시면 재현해볼 수 있는 최소한의 코드 내용과, 빌드에 사용하신 버전/환경 정보, `pyinstaller` 의 옵션 같은 정보도 부탁드립니다.
제 쪽에서 정확하게 확인하는데 도움이 될 것 같습니다.

공유주신 내용상으론, 말씀주신 것처럼 아래 부분에서 grpc 서버가 제대로 뜨지 못하고,
https://github.com/elbakramer/koapy/blob/701a6e5eadecafd28f32c3a7589d750e1822c4bc/koapy/backend/kiwoom_open_api_plus/pyside2/KiwoomOpenApiPlusManagerApplication.py#L298-L303

이후 클라이언트 커넥션 테스트 부분에서 실패하는 것처럼 보이네요.
https://github.com/elbakramer/koapy/blob/701a6e5eadecafd28f32c3a7589d750e1822c4bc/koapy/backend/kiwoom_open_api_plus/pyside2/KiwoomOpenApiPlusManagerApplication.py#L305-L315

앞의 `KiwoomOpenApiPlusServerApplicationProcess` 생성 과정에서 내부가 어떤식으로 처리가 되는지가 궁금한데..
가장 먼저 드는 생각은 subprocess 생성시 32bit python 을 못찾는다던가 그런건 아닌지 의심됩니다.
https://github.com/elbakramer/koapy/blob/701a6e5eadecafd28f32c3a7589d750e1822c4bc/koapy/backend/kiwoom_open_api_plus/pyside2/KiwoomOpenApiPlusManagerApplication.py#L68-L69
